backend/src/middleware/notifications/notificationsMiddleware.emails.spec.ts

```diff
@@ -0,0 +1,715 @@
+                },
+              ]),
+            },
+            errors: undefined,
+          })
+        })
+      })
+    })
+  })
+
+  describe('handleContentDataOfComment', () => {
+    describe('user comments post and author responds with in a comment and mentions the user', () => {
+      describe('all email notification settings are true', () => {
+        beforeEach(async () => {
+          authenticatedUser = await postAuthor.toJson()
+          await mutate({
+            mutation: createPostMutation,
+            variables: {
+              id: 'post',
+              title: 'This is the post',
+              content: `Hello, ${mentionString}, my trusty follower.`,
+              groupId: 'public-group',
+            },
+          })
+          authenticatedUser = await groupMember.toJson()
+          await mutate({
+            mutation: createCommentMutation,
+            variables: {
+              id: 'comment',
+              content: `Hello, my beloved author.`,
+              postId: 'post',
+            },
+          })
+          jest.clearAllMocks()
+          authenticatedUser = await postAuthor.toJson()
+          await mutate({
+            mutation: createCommentMutation,
+            variables: {
+              id: 'comment-2',
+              content: `Hello, ${mentionString}, my beloved follower.`,
+              postId: 'post',
+            },
+          })
+        })
+
+        it('sends only one email', () => {
+          expect(sendMailMock).toHaveBeenCalledTimes(1)
+        })
+
+        it('sends 2 notifications', async () => {
+          authenticatedUser = await groupMember.toJson()
+          await expect(
+            query({
+              query: notificationQuery,
+              variables: {
+                read: false,
+              },
+            }),
+          ).resolves.toMatchObject({
+            data: {
+              notifications: expect.arrayContaining([
+                {
+                  createdAt: expect.any(String),
+                  from: {
+                    __typename: 'Comment',
+                    id: 'comment-2',
+                    content:
+                      'Hello, <a class="mention" data-mention-id="group-member" href="/profile/group-member/group-member" target="_blank">@group-member</a>, my beloved follower.',
+                  },
+                  read: false,
+                  reason: 'commented_on_post',
+                  relatedUser: null,
+                },
+                {
+                  createdAt: expect.any(String),
+                  from: {
+                    __typename: 'Comment',
+                    id: 'comment-2',
+                    content:
+                      'Hello, <a class="mention" data-mention-id="group-member" href="/profile/group-member/group-member" target="_blank">@group-member</a>, my beloved follower.',
+                  },
+                  read: false,
+                  reason: 'commented_on_post',
```

Comment:
I am unsure if correct, but reading the rest of the tests this seems to be right

```suggestion
                  reason: 'mentioned_in_comment',
```